src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -1501,3 +1501,89 @@ ActStatus ActionHandlers::handle_s1_rel_req_during_attach(ControlBlock& cb)
 
     return ActStatus::PROCEED;
 }
+
+#define S10_FEATURE
+
+#ifdef S10_FEATURE
+
+/***************************************
+* Action handler : send_identification_request_to_old_mme
+***************************************/
+ActStatus ActionHandlers::send_identification_request_to_old_mme(ControlBlock& cb)
+{
+	log_msg(LOG_DEBUG, "Inside send_identification_request_to_old_mme");
+
+	UEContext *ue_ctxt = dynamic_cast<UEContext*>(cb.getPermDataBlock());
+	VERIFY_UE(cb, ue_ctxt, "Invalid UE");
+
+	struct ID_Q_msg id_msg;
+	id_msg.msg_type = identification_request;
+	id_msg.ue_idx = ue_ctxt->getContextID();
+
+	memcpy(&(id_msg._guti.plmn_id), &(ue_ctxt->getTai().tai_m.plmn_id), 3);
+	id_msg._guti.mme_grp_id = htons(mme_cfg->mme_group_id);
+	id_msg._guti.mme_code = mme_cfg->mme_code;
+    id_msg._guti.m_TMSI = htonl(ue_ctxt->getMTmsi());
+
+	cmn::ipc::IpcAddress destAddr;
+	destAddr.u32 = TipcServiceInstance::s10AppInstanceNum_c;
+
+    mmeStats::Instance()->increment(mmeStatsCounter::MME_MSG_TX_S10_IDENTIFICATION_REQUEST);
+	MmeIpcInterface &mmeIpcIf = static_cast<MmeIpcInterface&>(compDb.getComponent(MmeIpcInterfaceCompId));   
+	mmeIpcIf.dispatchIpcMsg((char *) &id_msg, sizeof(id_msg), destAddr);
+		
+	ProcedureStats::num_of_identification_req_sent++;
+	log_msg(LOG_DEBUG, "Leaving send_identification_request_to_old_mme ");
+	
+    return ActStatus::PROCEED;
+}
+
+/***************************************
+* Action handler : process_identification_response
+***************************************/
+ActStatus ActionHandlers::process_identification_response(ControlBlock& cb)
+{
+	log_msg(LOG_DEBUG, "Entering process_identification_response ");
+
+	UEContext *ue_ctxt = dynamic_cast<UEContext*>(cb.getPermDataBlock());
+	VERIFY_UE(cb, ue_ctxt, "Invalid UE");
+	
+	MmeAttachProcedureCtxt* procedure_p = dynamic_cast<MmeAttachProcedureCtxt*>(cb.getTempDataBlock());
+	VERIFY(procedure_p, return ActStatus::ABORT, "Procedure Context is NULL ");
+
+	MsgBuffer* msgBuf = static_cast<MsgBuffer*>(cb.getMsgData());
+	VERIFY(msgBuf, return ActStatus::ABORT, "Invalid message buffer ");
+	VERIFY(msgBuf->getLength() >= sizeof(struct id_resp_Q_msg), return ActStatus::ABORT, "Invalid CSRsp message length ");
+
+	const struct id_resp_Q_msg* id_resp_info = static_cast<const struct id_resp_Q_msg*>(msgBuf->getDataPointer());
+
+    /*if(id_resp_info->status != GTPV2C_CAUSE_REQUEST_ACCEPTED)
```

Comment:
Why this is commented ? What's plan when ID response is rejected.

Reply:
We are only considering the success scenarios as of now.  We are planning to implement the negative scenarios after the positive scenario code runs successfully.